gemma-core/src/main/java/ubic/gemma/core/loader/expression/singleCell/AbstractMexSingleCellDataLoaderConfigurer.java

```diff
@@ -56,16 +64,111 @@ public MexSingleCellDataLoader configureLoader( SingleCellDataLoaderConfig confi
                 }
             }
         }
-        MexSingleCellDataLoader loader = new MexSingleCellDataLoader( usedSampleNames, barcodeFiles, genesFiles, matrixFiles );
+
+        MexSingleCellDataLoader loader;
+        boolean apply10xFilter;
+        if ( config instanceof MexSingleCellDataLoaderConfig ) {
+            MexSingleCellDataLoaderConfig mexConfig = ( ( MexSingleCellDataLoaderConfig ) config );
+            if ( mexConfig.getApply10xFilter() != null ) {
+                apply10xFilter = mexConfig.getApply10xFilter();
+            } else {
+                apply10xFilter = detectUnfiltered10xData();
+            }
+        } else {
+            apply10xFilter = detectUnfiltered10xData();
+        }
+        if ( apply10xFilter ) {
+            SingleCell10xFilter filter = create10xFilter();
+            loader = createFilteredMexLoader( filter, usedSampleNames, sampleDirs );
+        } else {
+            loader = new MexSingleCellDataLoader( usedSampleNames, barcodeFiles, genesFiles, matrixFiles );
+        }
+
         if ( config instanceof MexSingleCellDataLoaderConfig ) {
             MexSingleCellDataLoaderConfig mexConfig = ( MexSingleCellDataLoaderConfig ) config;
             loader.setAllowMappingDesignElementsToGeneSymbols( mexConfig.isAllowMappingDesignElementsToGeneSymbols() );
             loader.setUseDoublePrecision( ( mexConfig.isUseDoublePrecision() ) );
         }
+
         return loader;
     }
 
     protected abstract List<String> getSampleNames();
 
     protected abstract List<Path> getSampleDirs();
+
+    /**
+     * Detect if the data is unfiltered 10x MEX data.
+     */
+    private boolean detectUnfiltered10xData() {
+        return false;
```

Comment:
This needs to be implemented.

There's a GEO-based configurer that has access to the sample metadata that could help cover more cases.

Otherwise detecting 10x data that is unfiltered might be difficult.